test/mocks/admin_api_handler.go

```diff
@@ -0,0 +1,149 @@
+package mocks
+
+import (
+	"net/http"
+	"sync/atomic"
+	"testing"
+)
+
+const defaultDBLessStatusResponse = `{
+	"version": "3.3.0",
+	"configuration": {
+		"database": "off",
+		"router_flavor": "traditional",
+		"role": "traditional",
+		"proxy_listeners": [
+			{
+				"backlog=%d+": false,
+				"ipv6only=on": false,
+				"ipv6only=off": false,
+				"ssl": false,
+				"so_keepalive=off": false,
+				"so_keepalive=%w*:%w*:%d*": false,
+				"listener": "0.0.0.0:8000",
+				"bind": false,
+				"port": 8000,
+				"deferred": false,
+				"so_keepalive=on": false,
+				"http2": false,
+				"proxy_protocol": false,
+				"ip": "0.0.0.0",
+				"reuseport": false
+			}
+		]
+	}
+}`
+
+// AdminAPIHandler is a mock implementation of the Admin API. It only implements the endpoints that are
+// required for the tests.
+type AdminAPIHandler struct {
+	mux *http.ServeMux
+	t   *testing.T
+
+	// ready is a flag that indicates whether the server should return a 200 OK or a 503 Service Unavailable.
+	// It's set to true by default.
+	ready bool
+
+	// workspaceExists makes `/workspace/workspaces/:id` return 200 when true, or 404 otherwise.
+	workspaceExists bool
+
+	// workspaceWasCreated is set to true when a workspace `POST /workspaces` was called.
+	workspaceWasCreated atomic.Bool
+}
+
+type AdminAPIHandlerOpt func(srv *AdminAPIHandler)
+
+func WithWorkspaceExists(exists bool) AdminAPIHandlerOpt {
+	return func(srv *AdminAPIHandler) {
+		srv.workspaceExists = exists
+	}
+}
+
+func WithReady(ready bool) AdminAPIHandlerOpt {
+	return func(srv *AdminAPIHandler) {
```

Comment:
Nit: since it's not a server but a handler I'd call the parameter `h` (or similar) here (and in other options)